packages/telegraf/extra/start_telegraf.sh

```diff
@@ -24,6 +24,14 @@ mkdir -p "${TELEGRAF_CONTAINERS_DIR}"
 # Migrate old containers dir to new location in case the cluster was upgraded.
 /opt/mesosphere/active/telegraf/tools/migrate_containers_dir.sh "${LEGACY_CONTAINERS_DIR}" "${TELEGRAF_CONTAINERS_DIR}"
 
+# Add symlinks to any additional configuration files in the user-editable Telegraf config directory
+shopt -s nullglob
+for file in ${TELEGRAF_EXTRA_CONFIG_DIR}*; do
+	ln -sfn $file ${TELEGRAF_CONFIG_DIR}$(basename "$file")
+done
```

Comment:
The intention was to create symlinks within `/opt/mesosphere/etc/telegraf/telegraf.d/` to the `/var/lib/dcos/telegraf/telegraf.d/` user-added config files, but I wasn't aware that we shouldn't be modifying `/opt/mesosphere`, even in a dcos startup script. Flipping all this logic and making the telegraf config dir point to the `/var/lib` path instead works, but still feels weird to me, and brings us back to [the original (similar) solution](https://github.com/dcos/dcos/pull/3572/files) which was to use the `/var/lib` dir and copy the `/opt/mesosphere` configs into it. Let me look more closely at modifying Telegraf itself to accept multiple dirs, that does seem like the cleanest solution.

Reply:
Yeah, nothing should be editing anything in `/opt/mesosphere/`, including DC/OS components aside from Pkgpanda. It's meant to be totally static apart from installs and upgrades.

FWIW we have at least one other package that copies config out of `/opt/mesosphere/` into a different location where it ultimately gets read from, so that users may edit it: https://github.com/dcos/dcos/blob/abaeb5cceedd5661b8d96ff47f8bb5ef212afbdc/packages/rexray/extra/dcos-rexray.service#L16

But I agree that it's not ideal.